src/utils/installPackage.js

```diff
@@ -72,19 +75,24 @@ async function installPeerDependencies(filepath, name, options) {
 }
 
 async function determinePackageManager(filepath) {
-  let configFile = await config.resolve(filepath, [
-    'yarn.lock',
-    'package-lock.json'
+  const [npmLockFile, yarnLockFile] = await Promise.all([
+    config.resolve(filepath, [PACKAGE_LOCK]),
+    config.resolve(filepath, [YARN_LOCK])
   ]);
-  let hasYarn = await checkForYarnCommand();
-
-  // If Yarn isn't available, or there is a package-lock.json file, use npm.
-  let configName = configFile && path.basename(configFile);
-  if (!hasYarn || configName === 'package-lock.json') {
+  const hasYarn = await checkForYarnCommand();
```

Comment:
This is no longer necessary, npm should be a fallback if we're using yarn. This slows down autoinstall

Reply:
On second thought this would only degrade performance if there aren't a lot of installs

Reply:
Doesn't it throw error if there is not yarn command and yarn.lock exists?

Reply:
well actually I have to include this to `Promise.all` then it shouldn't affect that much either

Reply:
Yeah, but if you catch the error and try again with npm if that happens it would be faster as that's what commandExists does, spawning processes is pretty expensive. But as the result is cached it probably doesn't matter that much you could however combine them. `if yarnLock => try yarn => catch => use npm & set hasYarn to false`